File: src/me/brendn/beta/gui/BetaTextRenderer.java
package me.brendn.beta.gui;

import me.brendn.minecraft.gui.TextRenderer;
import net.minecraft.client.Minecraft;

/**
 * TextRenderer implementation using Minecraft's FontRenderer class.
 */
public class BetaTextRenderer implements TextRenderer {

	private Minecraft mc;

	public BetaTextRenderer(Minecraft mc) {
		this.mc = mc;
	}

	@Override
	public void drawText(String text, int x, int y, int color) {
		mc.fontRenderer.drawString(text, x, y, color);
	}

	@Override
	public void drawTextWithShadow(String text, int x, int y, int color) {
		mc.fontRenderer.drawStringWithShadow(text, x, y, color);
	}

	@Override
	public int getWidth(String text) {
		return mc.fontRenderer.getStringWidth(text);
	}
}

File: src/main/kotlin/me/brendn/oak/mcbeta/gui/BetaTextRenderer.kt
package me.brendn.oak.mcbeta.gui

import me.brendn.oak.api.client.gui.TextRenderer
import net.minecraft.client.Minecraft

/**
 * TextRenderer implementation using Minecraft's FontRenderer class.
 */
class BetaTextRenderer(private val mc: Minecraft) : TextRenderer {

	override fun drawText(text: String, x: Int, y: Int, color: Int) {
		mc.fontRenderer.drawString(text, x, y, color)
	}

	override fun drawTextWithShadow(text: String, x: Int, y: Int, color: Int) {
		mc.fontRenderer.drawStringWithShadow(text, x, y, color)
	}

	override fun getWidth(text: String): Int {
		return mc.fontRenderer.getStringWidth(text)
	}
}